rules/do-you-securely-implement-and-handle-jwt-tokens/rule.md

```diff
@@ -0,0 +1,74 @@
+---
+type: rule
+title: Do you securely implement and handle JWT Tokens?
+uri: jwt-best-practices
+authors:
+  - title: Dhruv Mathur
+    url: https://www.ssw.com.au/people/dhruv-mathur/
+  - title: Gordon Beeming
+    url: https://www.ssw.com.au/people/gordon-beeming/
+created: 2024-03-06T13:42:55.753Z
+guid: fa645fa6-4054-448b-9b4b-d60a66422f1f
+---
+
+# Do you securely implement and handle JWT Tokens?
+
+JSON Web Tokens (JWTs) offer a compact and secure means of transmitting information between parties as JSON objects. Their ability to encode identity and claim information has made them popular in modern web development for authentication and authorization processes. However, handling JWTs securely is essential to prevent potential security breaches, such as token theft leading to unauthorized access.
+
+## JWTs Explained
+
+<https://www.youtube.com/embed/P2CPd9ynFLg>
+**Video**: JWTs eveyrthing you need to know (5mins)
+
+<!--endintro-->
+
+### Why JWTs Are Popular
+
+JWTs are favored for their:
+* **Compactness:** Making them easily transmissible via URL, POST parameters, or inside HTTP headers.
+* **Self-contained nature:** Carrying all necessary information, reducing the need for database lookups to authenticate requests.
+* **Versatility:** Being used in various scenarios, from Single Sign-On (SSO) to cross-service authentication in microservices architectures.
+
+### How JWT token are used
+
+JWTs token are used to implement authorised access for resouces (APIs) in an application, here is how a typical flow looks like:
+
+* **User Authentication:** A user logs in to an application, and the server creates and issues a signed JWT token upon successful authentication.
```

Comment:
Redundant use of token. Also technically the user signs in to an authority, or IDP, or token issuer; not an application.